source/base/fileinputoutput.cpp

```diff
@@ -317,12 +319,19 @@ IStream *NewIStream(const Path& p, unsigned int stype)
         throw POV_EXCEPTION(kCannotOpenFileErr, str);
     }
 
-    return new IFileStream(p().c_str());
+    stream = new IFileStream(p().c_str());
+    if (!(*stream))
+    {
+        delete stream;
+        return NULL;
+    }
+    return stream;
```

Comment:
It could be argued that the error is not in these functions, but in the callers' failure to test for errors. I think the proper first step to address this issue is to decide on, and formally specify, the intended behaviour of the functions; before that has happened, any code changes are moot.

Given that the functions already report some error situations by throwing an exception, it can be argued that the same mechanism should be used for other error situations as well.